the-events-calendar/event-tickets
# Feature/Changelogger

### 🎫 Ticket

No ticket 

### 🗒️ Description

See original PR [here](https://github.com/the-events-calendar/the-events-calendar/pull/4620) 

This PR seeks to mimic the above but for ET. Most of the code is just copied over, the only non-global changes are:
    - bin/class-tec-changelog-formatter.php -> file name can be changed to match the plugin
    - bin/class-tec-changelog-formatter.php L3 -> `* Jetpack Changelogger Formatter for The Events Calendar`
    - bin/class-tec-changelog-formatter.php L5 -> `* @package The Events Calendar`
    - bin/class-tec-changelog-formatter.php L14 -> `* Jetpack Changelogger Formatter for The Events Calendar`
    - bin/class-tec-changelog-formatter.php L16 -> `* Class TEC_Changelog_Formatter`
    - bin/class-tec-changelog-formatter.php L18 -> `class TEC_Changelog_Formatter extends Parser implements FormatterPlugin {`
    - bin/class-tec-changelog-formatter.php L40 ->`private $title = '# The Events Calendar Changelog';`
    - changelog.md -> entire file is plugin specific... I see in ET that it has a `changelog.txt` but not `.md` 
    - composer.json -> The lines vary so it's important to pay attention to which section you're adding them to. 
      - Also this line has tec in it: `"filename": "bin/class-tec-changelog-formatter.php"` -> make sure to change to whatever you changed the file to! 

Steps for testing
    1. Checkout PR branch `feature/changelogger`
    2. In terminal, run `npm install` -> this will add all the changes added in the new files
    3. Run `composer install` -> this will install the new dependencies added 
    4. Run `npm run changelog` -> This will create a new changelog file, with prompts for each required section.
    5. Look at the generated changelog in the new `/changelog` directory.
    6. Test adding a second entry and see that it makes another, separate file.
    7. Test generating the changelog with the
       command `./bin/process-changelog.sh 6.5.7 6.5.2 generate "2024-06-28"` -> `[file with command] [new version number] [old version number] [action type] [today's date]`
        - Worth noting that the readme.txt is only updated when the versions match 


### 🎥 Artifacts 

### ✔️ Checklist
- [ ] Changelog entry in the `readme.txt` file.
